src/pair_rann.cpp

```diff
@@ -0,0 +1,1247 @@
+				dlayersumy[jj][j]=0;
+				dlayersumz[jj][j]=0;
+				dssumx[jj][j]=0;
+				dssumy[jj][j]=0;
+				dssumz[jj][j]=0;
+				for (k=0;k<net1.dimensions[i];k++){
+					if (i==0&&j==0){
+						dlayerx[jj][k]=dfeaturesx[jj*f+k];
+						dlayery[jj][k]=dfeaturesy[jj*f+k];
+						dlayerz[jj][k]=dfeaturesz[jj*f+k];
+						dsx[jj][k]=-sx[jj*f+k];
+						dsy[jj][k]=-sy[jj*f+k];
+						dsz[jj][k]=-sz[jj*f+k];
+					}
+					double w1 = net1.Weights[i][j*net1.dimensions[i]+k];
+					dlayersumx[jj][j] += w1*dlayerx[jj][k];
+					dlayersumy[jj][j] += w1*dlayery[jj][k];
+					dlayersumz[jj][j] += w1*dlayerz[jj][k];
+					dssumx[jj][j] += w1*dsx[jj][k];
+					dssumy[jj][j] += w1*dsy[jj][k];
+					dssumz[jj][j] += w1*dsz[jj][k];
+				}
+				dlayersumx[jj][j] = dsum1*dlayersumx[jj][j];
+				dlayersumy[jj][j] = dsum1*dlayersumy[jj][j];
+				dlayersumz[jj][j] = dsum1*dlayersumz[jj][j];
+				dssumx[jj][j] *= dsum1;
+				dssumy[jj][j] *= dsum1;
+				dssumz[jj][j] *= dsum1;
+				if (i==L-1 && jj < (jnum-1)){
+//					j1 = jlist[jj];
+//					j1 &= NEIGHMASK;
+					j1 = jl[jj];
+					force[j1][0]+=dlayersumx[jj][j];
+					force[j1][1]+=dlayersumy[jj][j];
+					force[j1][2]+=dlayersumz[jj][j];
+					fm[j1][0]+=dssumx[jj][j];
+					fm[j1][1]+=dssumy[jj][j];
+					fm[j1][2]+=dssumz[jj][j];
+				}
+			}
+			if (i==L-1){
+				j1 = ilist[ii];
+				jj = jnum-1;
+				force[j1][0]+=dlayersumx[jj][j];
+				force[j1][1]+=dlayersumy[jj][j];
+				force[j1][2]+=dlayersumz[jj][j];
+				fm[j1][0]+=dssumx[jj][j];
+				fm[j1][1]+=dssumy[jj][j];
+				fm[j1][2]+=dssumz[jj][j];
+			}
+		}
+		//update values for next iteration
+		for (j=0;j<net1.dimensions[i+1];j++){
+			layer[j]=sum[j];
+			for (jj=0;jj<jnum;jj++){
+//				if (Bij[jj]==false)continue;
+				dlayerx[jj][j] = dlayersumx[jj][j];
+				dlayery[jj][j] = dlayersumy[jj][j];
+				dlayerz[jj][j] = dlayersumz[jj][j];
+				dsx[jj][j] = dssumx[jj][j];
+				dsy[jj][j] = dssumy[jj][j];
+				dsz[jj][j] = dssumz[jj][j];
+			}
+		}
+	}
+}
+
+
+
+//treats # as starting a comment to be ignored.
+int PairRANN::count_words(char *line){
+	int n = strlen(line) + 1;
+	char copy[n];
+	strncpy(copy,line,n);
+	char *ptr;
+	if ((ptr = strchr(copy,'#'))) *ptr = '\0';
+	if (strtok(copy," ,\t:_\n") == NULL) {
+		return 0;
+	}
+	n=1;
+	while ((strtok(NULL," ,\t:_\n"))) n++;
+	return n;
+}
+
+void PairRANN::init_list(int which, NeighList *ptr)
+{
+  listfull = ptr;
+}
+
+
+void PairRANN::init_style()
+{
+	  int irequest_full = neighbor->request(this,instance_me);
+	  neighbor->requests[irequest_full]->id = 1;
+	  neighbor->requests[irequest_full]->half = 0;
+	  neighbor->requests[irequest_full]->full = 1;
+}
+
+
+/* ----------------------------------------------------------------------
+   init for one type pair i,j and corresponding j,i
+------------------------------------------------------------------------- */
+
+double PairRANN::init_one(int i, int j)
+{
+  return cutmax;
+}
+
+void PairRANN::errorf(const char * message){
+	this->error->all(FLERR,message);
```

Comment:
please note that `Error::all()` may **only** be used for cases where **all** MPI ranks call it at the same time. Otherwise, `Error::one()` must be used or else LAMMPS will "hang".

Reply:
I've changed all calls to Error::one().